### PR TITLE
feat(icons): add the paint bucket icons

### PR DESCRIPTION
**Pull request by sallainCoveo**
### Proposed Changes
Main story: https://coveord.atlassian.net/browse/SPAAS-1112
subtask: https://coveord.atlassian.net/browse/SPAAS-1113

Only to add the paint bucket icons in Plasma, it will be useful for the builder!

### Potential Breaking Changes
No breaking changes

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [x] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/plasma/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

**Review by mikedidomizio — APPROVED** [review]

**Review by aiheon — APPROVED** [review]

**Comment by kdang-coveo** [comment]
Hey Sylvie, the paintbuckets do not respect the mod-classes being applied e.g:
CleanShot 2022-02-11 at 10 51 29@2x

CleanShot 2022-02-11 at 10 51 41@2x

I think they should respect the mod-classes, we might need to ask UX to revise these svgs unless its okay for just the same stroke colour

**Comment by sallainCoveo** [comment]
> Hey Sylvie, the paintbuckets do not respect the mod-classes being applied e.g: CleanShot 2022-02-11 at 10 51 29@2x
> 
> CleanShot 2022-02-11 at 10 51 41@2x
> 
> I think they should respect the mod-classes, we might need to ask UX to revise these svgs unless its okay for just the same stroke colour

Oh no! I don't think it will have an impact on the current project, but still I think it's worth looking into it before we merge those ones. @johnnyCoveo what do you think about it?

File: packages/style/resources/icons/svg/paint-bucket-16px.svg [hunk 1]
```diff
@@ -0,0 +1,8 @@
...
+</mask>
+<path d="M5.37018 7.42039C5.37018 7.97268 5.8179 8.42039 6.37018 8.42039C6.92247 8.42039 7.37018 7.97268 7.37018 7.42039H5.37018ZM2.40723 5.5653C2.40723 6.11758 2.85494 6.5653 3.40723 6.5653C3.95951 6.5653 4.40723 6.11758 4.40723 5.5653H2.40723ZM7.37018 7.42039V1.48408H5.37018V7.42039H7.37018ZM7.37018 1.48408C7.37018 0.117204 6.25746 -1 4.88871 -1V1C5.14958 1 5.37018 1.21847 5.37018 1.48408H7.37018ZM4.88871 -1C3.51996 -1 2.40723 0.117204 2.40723 1.48408H4.40723C4.40723 1.21847 4.62783 1 4.88871 1V-1ZM2.40723 1.48408V5.5653H4.40723V1.48408H2.40723Z" fill="#282829" mask="url(#path-2-inside-1_194_5423)"/>
+<path d="M15.5 14.1449C15.5 14.9084 14.9082 15.5 14.1481 15.5C13.388 15.5 12.7963 14.9084 12.7963 14.1449C12.7963 13.9919 12.8756 13.6876 13.0495 13.2556C13.2145 12.8455 13.4378 12.3838 13.6658 11.9453C13.8332 11.6235 14.0008 11.3183 14.146 11.061C14.292 11.322 14.4611 11.632 14.6299 11.9583C14.8581 12.3995 15.0817 12.8629 15.247 13.2719C15.4219 13.7045 15.5 14.0026 15.5 14.1449Z" stroke="#282829" stroke-miterlimit="10" stroke-linecap="round" stroke-linejoin="round"/>
+</svg>
```
Comment (gdostie) [inline packages/style/resources/icons/svg/paint-bucket-16px.svg:8]: Move the `stoke` attributes to the root `svg` tag plz

**Review by gdostie — CHANGES_REQUESTED** [review]
We cannot leave strokes directly on paths, otherwise we won't be able to apply colors on the icons with css

File: packages/style/resources/icons/svg/paint-bucket-16px.svg [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+</mask>
+<path d="M5.37018 7.42039C5.37018 7.97268 5.8179 8.42039 6.37018 8.42039C6.92247 8.42039 7.37018 7.97268 7.37018 7.42039H5.37018ZM2.40723 5.5653C2.40723 6.11758 2.85494 6.5653 3.40723 6.5653C3.95951 6.5653 4.40723 6.11758 4.40723 5.5653H2.40723ZM7.37018 7.42039V1.48408H5.37018V7.42039H7.37018ZM7.37018 1.48408C7.37018 0.117204 6.25746 -1 4.88871 -1V1C5.14958 1 5.37018 1.21847 5.37018 1.48408H7.37018ZM4.88871 -1C3.51996 -1 2.40723 0.117204 2.40723 1.48408H4.40723C4.40723 1.21847 4.62783 1 4.88871 1V-1ZM2.40723 1.48408V5.5653H4.40723V1.48408H2.40723Z" fill="#282829" mask="url(#path-2-inside-1_194_5423)"/>
+<path d="M15.5 14.1449C15.5 14.9084 14.9082 15.5 14.1481 15.5C13.388 15.5 12.7963 14.9084 12.7963 14.1449C12.7963 13.9919 12.8756 13.6876 13.0495 13.2556C13.2145 12.8455 13.4378 12.3838 13.6658 11.9453C13.8332 11.6235 14.0008 11.3183 14.146 11.061C14.292 11.322 14.4611 11.632 14.6299 11.9583C14.8581 12.3995 15.0817 12.8629 15.247 13.2719C15.4219 13.7045 15.5 14.0026 15.5 14.1449Z" stroke="#282829" stroke-miterlimit="10" stroke-linecap="round" stroke-linejoin="round"/>
```
Comment (gdostie) [inline packages/style/resources/icons/svg/paint-bucket-16px.svg:7]:
```suggestion
<path d="M15.5 14.1449C15.5 14.9084 14.9082 15.5 14.1481 15.5C13.388 15.5 12.7963 14.9084 12.7963 14.1449C12.7963 13.9919 12.8756 13.6876 13.0495 13.2556C13.2145 12.8455 13.4378 12.3838 13.6658 11.9453C13.8332 11.6235 14.0008 11.3183 14.146 11.061C14.292 11.322 14.4611 11.632 14.6299 11.9583C14.8581 12.3995 15.0817 12.8629 15.247 13.2719C15.4219 13.7045 15.5 14.0026 15.5 14.1449Z" stroke-miterlimit="10" stroke-linecap="round" stroke-linejoin="round"/>
```

File: packages/style/resources/icons/svg/paint-bucket-24px.svg [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+</mask>
+<path d="M8.96273 11.2755C8.96273 11.8278 9.41045 12.2755 9.96273 12.2755C10.515 12.2755 10.9627 11.8278 10.9627 11.2755H8.96273ZM5.25903 8.95662C5.25903 9.5089 5.70675 9.95662 6.25903 9.95662C6.81132 9.95662 7.25903 9.5089 7.25903 8.95662H5.25903ZM10.9627 11.2755V3.8551H8.96273V11.2755H10.9627ZM10.9627 3.8551C10.9627 2.28416 9.68334 1 8.11088 1V3C8.57546 3 8.96273 3.38543 8.96273 3.8551H10.9627ZM8.11088 1C6.53843 1 5.25903 2.28416 5.25903 3.8551H7.25903C7.25903 3.38543 7.6463 3 8.11088 3V1ZM5.25903 3.8551V8.95662H7.25903V3.8551H5.25903Z" fill="#282829" mask="url(#path-2-inside-1_194_5428)"/>
+<path d="M21.5 19.6811C21.5 20.7044 20.7045 21.5 19.6852 21.5C18.6658 21.5 17.8704 20.7044 17.8704 19.6811C17.8704 19.4632 17.9786 19.0627 18.1959 18.5229C18.4044 18.0049 18.6855 17.4238 18.9714 16.874C19.2249 16.3864 19.4789 15.9296 19.683 15.5723C19.8878 15.934 20.1434 16.3973 20.3985 16.8905C20.6845 17.4435 20.966 18.0266 21.1747 18.5431C21.3929 19.083 21.5 19.4762 21.5 19.6811Z" stroke="#282829" stroke-miterlimit="10" stroke-linecap="round" stroke-linejoin="round"/>
```
Comment (gdostie) [inline packages/style/resources/icons/svg/paint-bucket-24px.svg:7]:
```suggestion
<path d="M21.5 19.6811C21.5 20.7044 20.7045 21.5 19.6852 21.5C18.6658 21.5 17.8704 20.7044 17.8704 19.6811C17.8704 19.4632 17.9786 19.0627 18.1959 18.5229C18.4044 18.0049 18.6855 17.4238 18.9714 16.874C19.2249 16.3864 19.4789 15.9296 19.683 15.5723C19.8878 15.934 20.1434 16.3973 20.3985 16.8905C20.6845 17.4435 20.966 18.0266 21.1747 18.5431C21.3929 19.083 21.5 19.4762 21.5 19.6811Z" stroke-miterlimit="10" stroke-linecap="round" stroke-linejoin="round"/>
```

File: packages/style/resources/icons/svg/paint-bucket-32px.svg [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+</mask>
+<path d="M12.1478 14.9857C12.1478 15.538 12.5955 15.9857 13.1478 15.9857C13.7001 15.9857 14.1478 15.538 14.1478 14.9857H12.1478ZM6.96265 11.7393C6.96265 12.2916 7.41036 12.7393 7.96265 12.7393C8.51493 12.7393 8.96265 12.2916 8.96265 11.7393H6.96265ZM14.1478 14.9857V4.59714H12.1478V14.9857H14.1478ZM14.1478 4.59714C14.1478 2.61808 12.5351 1 10.5552 1V3C11.4272 3 12.1478 3.71934 12.1478 4.59714H14.1478ZM10.5552 1C8.57537 1 6.96265 2.61808 6.96265 4.59714H8.96265C8.96265 3.71934 9.68325 3 10.5552 3V1ZM6.96265 4.59714V11.7393H8.96265V4.59714H6.96265Z" fill="#282829" mask="url(#path-2-inside-1_194_5433)"/>
+<path d="M29.5 26.7536C29.5 28.2963 28.2971 29.5 26.7593 29.5C25.2214 29.5 24.0186 28.2963 24.0186 26.7536C24.0186 26.4058 24.1847 25.8128 24.4888 25.0574C24.784 24.3238 25.181 23.5038 25.5825 22.7314C25.9833 21.9605 26.3847 21.2448 26.6862 20.7213C26.7105 20.6791 26.7341 20.6382 26.757 20.5987C26.781 20.6405 26.8058 20.6839 26.8313 20.7286C27.133 21.2574 27.5345 21.9795 27.9355 22.7548C28.3373 23.5316 28.7346 24.3542 29.0301 25.0854C29.3351 25.8401 29.5 26.4234 29.5 26.7536Z" stroke="#282829" stroke-miterlimit="10" stroke-linecap="round" stroke-linejoin="round"/>
```
Comment (gdostie) [inline packages/style/resources/icons/svg/paint-bucket-32px.svg:7]:
```suggestion
<path d="M29.5 26.7536C29.5 28.2963 28.2971 29.5 26.7593 29.5C25.2214 29.5 24.0186 28.2963 24.0186 26.7536C24.0186 26.4058 24.1847 25.8128 24.4888 25.0574C24.784 24.3238 25.181 23.5038 25.5825 22.7314C25.9833 21.9605 26.3847 21.2448 26.6862 20.7213C26.7105 20.6791 26.7341 20.6382 26.757 20.5987C26.781 20.6405 26.8058 20.6839 26.8313 20.7286C27.133 21.2574 27.5345 21.9795 27.9355 22.7548C28.3373 23.5316 28.7346 24.3542 29.0301 25.0854C29.3351 25.8401 29.5 26.4234 29.5 26.7536Z" stroke-miterlimit="10" stroke-linecap="round" stroke-linejoin="round"/>
```

**Review by gdostie — CHANGES_REQUESTED** [review]
No stroke attribute on the paths directly please 😄

**Review by kdang-coveo — APPROVED** [review]

File: packages/style/resources/icons/svg/paint-bucket-16px.svg [hunk 5]
```diff
@@ -0,0 +1,5 @@
+<svg viewBox="0 0 16 16" fill="none" stroke="black" xmlns="http://www.w3.org/2000/svg">
```
Comment (gdostie) [inline packages/style/resources/icons/svg/paint-bucket-16px.svg:1]: All of our other icons have grey-100 has a default color

```suggestion
<svg viewBox="0 0 16 16" fill="none" stroke="#282829" xmlns="http://www.w3.org/2000/svg">
```

File: packages/style/resources/icons/svg/paint-bucket-24px.svg [hunk 6]
```diff
@@ -0,0 +1,5 @@
+<svg viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg" stroke="black">
```
Comment (gdostie) [inline packages/style/resources/icons/svg/paint-bucket-24px.svg:1]:
```suggestion
<svg viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg" stroke="#282829">
```

File: packages/style/resources/icons/svg/paint-bucket-32px.svg [hunk 7]
```diff
@@ -0,0 +1,5 @@
+<svg viewBox="0 0 32 32" fill="none" xmlns="http://www.w3.org/2000/svg" stroke="black">
```
Comment (gdostie) [inline packages/style/resources/icons/svg/paint-bucket-32px.svg:1]:
```suggestion
<svg viewBox="0 0 32 32" fill="none" xmlns="http://www.w3.org/2000/svg" stroke="#282829">
```

**Review by gdostie — APPROVED** [review]

**Pull request merged**